lib/easybill-rest-client/generic_api.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+module EasybillRestClient
+  module GenericApi
+    def self.new(resource_name, resource_class)
```

Comment:
I also thought about `Struct.new(:foo, :bar)`. But that's different, because it's the interface exposed by the stdlib. Also the correct way to add methods is this:

```ruby
Customer = Struct.new(:name, :address) do
  def greeting
    "Hello #{name}!"
  end
end
```

not

```ruby
class Customer < Struct.new(:name, :address)
  def greeting
    "Hello #{name}!"
  end
end
```

thus avoiding the anonymous class and multiple inheritance levels. But adding the block syntax just adds more complexity, that's why I'd just go with the simple solution here.